unic/utils/src/macros.rs

```diff
@@ -0,0 +1,307 @@
+/// Macro for declaring a character property.
+///
+/// # Syntax
+///
+/// ```
+/// # #[macro_use] extern crate unic_utils;
+/// # fn main() {}
+/// char_property! {
+///     /// Zero or more attributes
+///     pub enum PropertyName {
+///         /// Exactly one attribute
+///         VariantName {
+///             abbr => AbbrName,
+///             long => Long_Name,
+///             display => "&'static str that is a nicer presentation of the name",
+///         }
+///
+///         /// All annotations on the variant are optional
+///         Variant2 {}
+///     }
+///
+///     /// Zero or more attributes
+///     pub mod abbr_names;
+/// }
+/// ```
+///
+/// # Effect
+///
+/// - Implements `CharProperty` with the `abbr` and `long` presented in the appropriate method
+/// - Implements `FromStr` accepting the variant name or the abbr or long forms
+/// - Implements `Display` using the given string, falling back when not provided on
+///   the long name, the short name, and the variant name, in that order
+/// - Populates the module `abbr_names` with `pub use` bindings of variants to their short names
```

Comment:
In some applications, I expect `FromStr` would be preferred over including more symbols. But since it is here, it does make sense to expose it for when it is useful.